index.bs

```diff
@@ -339,6 +414,31 @@ The {{DeviceMotionEvent/interval}} attribute must be initialized with the interv
 
 Implementations that are unable to provide all attributes must initialize the values of the unknown attributes to null. If an implementation can never provide motion information, the event should be fired with the {{DeviceMotionEvent/acceleration}}, {{DeviceMotionEvent/accelerationIncludingGravity}} and {{DeviceMotionEvent/rotationRate}} attributes set to null.
 
+<h3 id="id=permission-model">Permission model</h3>
+
+<p>Implementations may choose to share device orientation &amp; motion only if the
+user (or user agent on behalf of the user) has granted <dfn>permission</dfn>.
+The <a>permission</a> to share device orientation &amp; motion
+for a given <a for=/>origin</a> is one of three strings:
```

Comment:
As mentioned in my comments on [#57](https://github.com/w3c/deviceorientation/issues/57#issuecomment-447126594), to prevent divergence between specifications can we define this in terms of the same ["accelerometer"](https://www.w3.org/TR/permissions/#accelerometer), ["gyroscope"](https://www.w3.org/TR/permissions/#gyroscope) and ["magnetometer"](https://www.w3.org/TR/permissions/#magnetometer) permissions out of the Permissions API specification that are checked by the Generic Sensors family of orientation sensors?

Implementations don't have to actually track these permissions separately (granting one would grant them all) but it means that developers transitioning between the two APIs don't have to deal with separate permissions systems.

Reply:
Sorry, I did not understand this feedback. What's the concrete change proposed here?

Without understanding your proposal, all I can say is that I find the terms "accelerometer", "gyroscope" and "magnetometer" confusing when it comes to this specification. As in it is not obvious which string relates to which event.

Reply:
First let's make sure we understand what the sensors are and where their data come from:

The **`"deviceorientation"` event** provides the same data from the platform as a [`RelativeOrientationSensor`](https://www.w3.org/TR/orientation-sensor/#relativeorientationsensor-model) and so I am suggesting that it follow the same [permission model](https://www.w3.org/TR/orientation-sensor/#model) as the one defined in the new spec. That is, requesting access to this sensor should require obtaining the `"accelerometer"` and `"gyroscope"` permission tokens. This reflects the fact that it is the device's low-level accelerometer and gyroscope which are combined (through sensor fusion) into the high-level device orientation sensor which this event exposes to the web platform.

The **`"devicemotion"` event** provides the same data from the platform as an [`Accelerometer`](https://www.w3.org/TR/accelerometer/#accelerometer-interface), [`LinearAccelerationSensor`](https://www.w3.org/TR/accelerometer/#linearaccelerationsensor-interface) and [`Gyroscope`](https://www.w3.org/TR/gyroscope/#gyroscope-interface) used simultaneously. The permission models for these sensors also require the page to obtain the `"accelerometer"` and `"gyroscope"` permission tokens as this event directly provides the data from these low-level sensors.

My proposal is that the `DeviceMotionEvent.requestPermission()` and `DeviceOrientationEvent.requestPermission()` functions are specified as simultaneously requesting the `"accelerometer"` and `"gyroscope"` permission tokens.

In addition, the **`"deviceorientationabsolute"` event** provides the same data from the platform as an [`AbsoluteOrientationSensor`](https://www.w3.org/TR/orientation-sensor/#absoluteorientationsensor-model) which has the additional requirement of the page obtaining the `"magnetometer"` permission token in addition to `"accelerometer"` and `"gyroscope"` because this sensor fusion uses the device compass to determine the absolute orientation relative to magnetic north.

I therefore propose, since there is no `DeviceAbsoluteOrientationEvent` on which to add another  `requestPermission()` function, that the `DeviceOrientationEvent.requestPermission()` function take an optional `absolute` boolean parameter (defaulting to `false`) which causes it to also request the `"magnetometer"` permission token.

Most developers won't care about the details behind these methods but the benefit of defining these permissions consistently between specifications is that who do want to transition between these APIs don't have to request to do effectively the same thing in two different ways.

Reply:
It seems like this is purely a matter of spec wording and would not have any behavior change on JS or native side. If people want to this, I am not opposing although I personally have no interest in going in this direction given that the other specs you mention are not supported in WebKit.

If people want to align with the other sensor specs, could someone else please take over this PR?

Reply:
@reillyeon I believe the text you proposed could be a very helpful informative note in the spec aimed at implementers who ship both this API and the related Generic Sensor-based APIs.

With my chair hat on, I'd suggest that informative text to be baked in a separate PR, perhaps authored by you.

It would be preferred to not expand this PR's scope as to allow @cdumez remain the owner of this PR.

Reply:
I agree. We can build alignment between the specifications in follow-up.

Reply:
@reillyeon maybe it’d be good to open a new issue with your proposed text to solicit feedback in parallel to this PR in being baked? Related: https://w3c.github.io/permissions/#permission-registry